neuralpredictors/training/utils.py

```diff
@@ -0,0 +1,20 @@
+import collections
+from typing import Dict
+
+
+def deep_update(d: Dict, u: Dict) -> Dict:
```

Comment:
I'd say `Mapping` would be the correct type for all parameters and the return value here because your code will work if `d` and `u` are of type `Mapping` and there is no guarantee that the return value will be of type `Dict`.

Reply:
Or even more correct would be:
```
import collections
from typing import Dict, TypeVar

T = TypeVar("T", bound=collections.abc.Mapping)

def deep_update(d: T, u: collections.abc.Mapping) -> T:
    ...
```
This would tell a static type checker that the return value has to be of the same type as `d` and that `d` has to be a subclass of `Mapping`.